dll/ntdll/ldr/ldrapi.c

```diff
@@ -1175,12 +1176,13 @@ LdrEnumerateLoadedModules(
 
     /* Release loader lock */
     Status = LdrUnlockLoaderLock(0, Cookie);
-    ASSERT(NT_SUCCESS(Status));
 
     /* Reset any successful status to STATUS_SUCCESS,
      * but leave failure to the caller */
     if (NT_SUCCESS(Status))
         Status = STATUS_SUCCESS;
+    else
+        ASSERT(NT_SUCCESS(Status));
```

Comment:
This changes behavior in debug builds, and this change does not seem like a good change to me.